kernel/kernel-api/src/main/java/io/delta/kernel/metrics/SnapshotReport.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.metrics;
+
+import java.util.Optional;
+
+/** Defines the metadata and metrics for a snapshot construction {@link MetricsReport} */
+public interface SnapshotReport extends DeltaOperationReport {
+
+  /**
+   * For a time-travel by version query, this is the version provided. For a time-travel by
+   * timestamp query, this is the version resolved from the provided timestamp. For a latest
+   * snapshot, this is the version read from the delta log.
+   *
+   * <p>This is empty when this report is for a failed snapshot construction, and the error occurs
+   * before a version can be resolved.
+   *
+   * @return the version of the snapshot
+   */
+  Optional<Long> version();
+
+  /**
+   * @return the timestamp provided for time-travel, empty if this is not a timestamp-based
+   *     time-travel query
+   */
+  Optional<Long> providedTimestamp();
+
+  /** @return the metrics for this snapshot construction */
+  SnapshotMetricsResult snapshotMetrics();
+
+  @Override
+  default String operationType() {
+    return DeltaOperationReport.OperationType.SNAPSHOT.toString();
```

Comment:
Hm. If we return `String` still I wonder if the enum isn't as useful? what if we returned the enum value instead?

i.e. how do we expect this `operationType()` method to be used?

I'm fine with whatever you decide, no need to block.

Reply:
My thought is that it's useful to go to `DeltaOperationType.OperationType` and see all the possible operation types we might encounter. I think it makes sense to be string instead of the enum to try to keep the metric return types to standard built in types.

Reply:
> My thought is that it's useful to go to DeltaOperationReport.OperationType and see all the possible operation types we might encounter.

But there's nothing enforcing this, right? It's just declaring an enum -- children of `DeltaOperationReport` don't actually _need_ to add a new type to this enum, they could just return an arbitrary string for `String operationType()`.

If we had `operationType` return an enum, then it's fair to require and expect that children will add new values for the enum.

Reply:
> But maybe the answer to that is we don't need the enum and it's possible to just look at all the metrics interfaces we have added.

I think the question is: how will the result of `operationType` be used? is it just for printing? do we want to check for a particular value (in this case, an enum would be nice)

Reply:
Hmm. I think we would probably wouldn't check for a specific value. If you want a snapshot report you would check if it's an instance of `SnapshotReport`.

I think i'll remove the enum and keep `operationType` to return a string.

Reply:
SGTM!